### DISCUSSION TITLE
[@mui/lab/DateRangePicker] Add to Doc

### DISCUSSION THREAD

**Discussion by aress31** [react-hook-form/react-hook-form#6906] [Q&A] [answered]
I am struggling to integrate `import { LocalizationProvider, DateRangePicker } from "@mui/lab";` with RHF. If you could please add an example on the RHF doc, that would help a lot. I namely need to control/register the `DateRangePicker` component to be able to get the `start/end date`.

**Comment by bluebill1049** [comment]
please provide a base codesandbox.

**Comment by aress31** [comment]
A code sandbox is what I am asking for. Guidance on how to integrate RHF, with DateRangePicker.

**Reply by lundn** [reply]
You should provide a basic codesandbox with a simplified setup based on your code. It's easier for the developer and others to help out.

**Comment by aress31** [comment]
What I am requesting is an additional example to the doc.

**Reply by lundn** [reply]
Could you please show us a snippet of your current, non-working implementation? Your code within `<LocalizationProvider dateAdapter={AdapterDateFns}>`?

**Reply by aress31** [reply]
Certainly, I tried these two different approaches:
```js
const renderGeneralInformation = (control, formState, register) => (
  <FormGroup>
    --- SNIP ---
    <LocalizationProvider dateAdapter={AdapterDateFns}>
      <Controller
        control={control}
        name="date_range"
        render={(props) => (
          <DateRangePicker
            startText="Start"
            endText="End"
            value={props.value}
            onChange={(e) => props.onChange(e)}
            renderInput={(startProps, endProps) => (
              <React.Fragment>
                <TextField {...startProps} sx={{ marginTop: 2 }} />
                <Box sx={{ mx: 2 }}> to </Box>
                <TextField {...endProps} />
              </React.Fragment>
            )}
          />
        )}
      />
    </LocalizationProvider>
  </FormGroup>
);
```

```js
const renderGeneralInformation = (control, formState, register) => (
  <FormGroup>
    --- SNIP ---
    <LocalizationProvider dateAdapter={AdapterDateFns}>
          <DateRangePicker
            startText="Start"
            endText="End"
            {register("date_range")}
            renderInput={(startProps, endProps) => (
              <React.Fragment>
                <TextField {...startProps} sx={{ marginTop: 2 }} />
                <Box sx={{ mx: 2 }}> to </Box>
                <TextField {...endProps} />
              </React.Fragment>
            )}
          />
    </LocalizationProvider>
  </FormGroup>
);
```

Unfortunately, none of them work, just throwing some JS errors. In the current doc there is an example on how to integrate the `DatePicker` component but not the `DateRangePicker`, it seems that a different approach should be taken to integrate this component with `RHF`.

**Reply by lundn** [reply]
Could you try something like this:
```
<LocalizationProvider dateAdapter={AdapterDateFns}>
    <Controller
      control={control}
      name={"name"}
      render={({ field: { onChange, value, ref } }) => {
        return (
          <DateRangePicker
            value={value}
            onChange={(newValue) => {
              onChange(newValue);
            }}
            inputRef={ref}
            renderInput={(startProps, endProps) => (
               ...
            )}
          />
        );
      }}
    />
  </LocalizationProvider>
```

**Reply by aress31** [reply]
@lundn, I am getting the following error message:

![image](https://user-images.githubusercontent.com/11601622/139127249-dee4aab7-ef4a-4bd1-96da-5e85ae248079.png)

**Comment by lundn** [comment, marked as answer]
@aress31 Again, it's hard to pinpoint your errors without a minimalistic codesandbox. 

I got this one working, please check it out.
Check this: https://codesandbox.io/s/staticdaterangepickerdemo-material-demo-forked-29wk4

**Reply by aress31** [reply]
Thanks so much I was missing the default initialisation. IMHO, this example is worth adding to the doc!

**Reply by lundn** [reply]
Your welcome! Glad we managed to solve it!

**Reply by bluebill1049** [reply]
thank you very much @lundn
